platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/sync/SingleNodeSyncGossip.java

```diff
@@ -0,0 +1,255 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.gossip.sync;
+
+import static com.swirlds.logging.LogMarker.RECONNECT;
+import static com.swirlds.platform.SwirldsPlatform.PLATFORM_THREAD_POOL_NAME;
+
+import com.swirlds.common.config.BasicConfig;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.merkle.synchronization.config.ReconnectConfig;
+import com.swirlds.common.system.NodeId;
+import com.swirlds.common.system.SoftwareVersion;
+import com.swirlds.common.system.address.AddressBook;
+import com.swirlds.common.threading.framework.QueueThread;
+import com.swirlds.common.threading.framework.StoppableThread;
+import com.swirlds.common.threading.framework.config.StoppableThreadConfiguration;
+import com.swirlds.common.threading.interrupt.InterruptableConsumer;
+import com.swirlds.common.threading.manager.ThreadManager;
+import com.swirlds.common.threading.pool.ParallelExecutor;
+import com.swirlds.common.utility.Clearable;
+import com.swirlds.common.utility.LoggingClearables;
+import com.swirlds.platform.Crypto;
+import com.swirlds.platform.FreezeManager;
+import com.swirlds.platform.PlatformConstructor;
+import com.swirlds.platform.StartUpEventFrozenManager;
+import com.swirlds.platform.components.CriticalQuorum;
+import com.swirlds.platform.components.CriticalQuorumImpl;
+import com.swirlds.platform.components.EventMapper;
+import com.swirlds.platform.components.state.StateManagementComponent;
+import com.swirlds.platform.event.EventIntakeTask;
+import com.swirlds.platform.gossip.AbstractGossip;
+import com.swirlds.platform.gossip.FallenBehindManagerImpl;
+import com.swirlds.platform.gossip.shadowgraph.ShadowGraph;
+import com.swirlds.platform.gossip.shadowgraph.SingleNodeNetworkSync;
+import com.swirlds.platform.metrics.EventIntakeMetrics;
+import com.swirlds.platform.observers.EventObserverDispatcher;
+import com.swirlds.platform.state.SwirldStateManager;
+import com.swirlds.platform.state.signed.SignedState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Sync gossip using the protocol negotiator.
+ */
+public class SingleNodeSyncGossip extends AbstractGossip {
+
+    private static final Logger logger = LogManager.getLogger(SingleNodeSyncGossip.class);
+
+    private final InterruptableConsumer<EventIntakeTask> eventIntakeLambda;
+    private final Clearable clearAllPipelines;
+
+    /**
+     * A list of threads that execute the sync protocol using bidirectional connections
```

Comment:
This comment isn't accurate for a single node network: this thread simply calls `eventTaskCreator::createEvent` in a loops, with no connections involved

Reply:
removed javadoc